sdk/dotnet/Pulumi/Deployment/Deployment_Prepare.cs

```diff
@@ -40,8 +40,12 @@ public partial class Deployment
             // If no parent was provided, parent to the root resource.
             LogExcessive($"Getting parent urn: t={type}, name={name}, custom={custom}, remote={remote}");
             var parentUrn = options.Parent != null
-                ? await options.Parent.Urn.GetValueAsync().ConfigureAwait(false)
-                : await GetRootResourceAsync(type).ConfigureAwait(false);
+                ? await options.Parent.Urn.GetValueAsync(whenUnknown: "").ConfigureAwait(false)
```

Comment:
This seemed like a reasonable thing to default to branch 2 when options.Parent is not null but unknown; however I'm not sure. Perhaps better to revert to original behavior simply with `default!`.